test/dynamo/test_bridge.py

```diff
@@ -233,23 +243,28 @@ def __init__(self):
     module = Emb()
     module.to(device)
 
-    @torch.compile(backend="openxla_eval")
     def foo(x):
       return module(x)
 
     x = torch.randint(0, 10, (10,), device=device)
-    foo(x)
+    self._compile_and_check(foo, (x,), backend="openxla_eval")
 
   def test_inputs_not_computed(self):
 
-    @torch.compile(backend="openxla")
     def foo(x):
       return x * 2
 
     device = xm.xla_device()
     x = torch.rand(5, device=device)
     x = x.unsqueeze(dim=-1)
-    foo(x)
+    self._compile_and_check(foo, (x,))
+
+  def test_factory_copy(self):
+
+    def foo(device):
+      return torch.arange(5, device="cpu").to(device)
+
+    self._compile_and_check(foo, (xm.xla_device(),))
```

Comment:
will this case fail without the `XLAConstructorMoverPass`?

Reply:
Yes. The partitioner will split that function into 2: (i) arange; and (ii) _to_copy. The first one will be returning a CPU tensor, which will fail.